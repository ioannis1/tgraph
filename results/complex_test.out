CREATE EXTENSION complex;
SELECT '(1,2')::complex;
ERROR:  syntax error at or near ")"
LINE 1: SELECT '(1,2')::complex;
                     ^
SELECT '(1,2)'::complex +   '(1,2)';
 ?column? 
----------
 (2,4)
(1 row)

SELECT '(1,2)'::complex -   '(1,2)';
 ?column? 
----------
 (0,0)
(1 row)

SELECT '(1,2)'::complex =   '(1,2)';
 ?column? 
----------
 t
(1 row)

SELECT '(1,2)'::complex >   '(0,2)';
 ?column? 
----------
 t
(1 row)

SELECT '(1,2)'::complex >=  '(8,2)';
 ?column? 
----------
 f
(1 row)

SELECT mag( '(3,4)'::complex );
 mag 
-----
   5
(1 row)

SELECT complex_mag_squared( '(3,4)'::complex );
 complex_mag_squared 
---------------------
                  25
(1 row)

SELECT  2 > '(3,4)'::complex;
 ?column? 
----------
 f
(1 row)

SELECT 25 >= '(3,4)'::complex;
 ?column? 
----------
 t
(1 row)

SELECT  ('(1,8)'::complex@);      -- complex conjugate
 ?column? 
----------
 (1,-8)
(1 row)

SELECT |'(3,4)'::complex ;        -- magnitude
 ?column? 
----------
        5
(1 row)

SELECT complex_new(1,8);
 complex_new 
-------------
 (0,0)
(1 row)

SELECT complex_new(3,1);
 complex_new 
-------------
 (0,0)
(1 row)

SELECT complex_real('(-2,3)'::complex);
 complex_real 
--------------
            0
(1 row)

SELECT complex_img('(2,-3)'::complex);
 complex_img 
-------------
           0
(1 row)

SELECT complex_overlaps('(2,2)'::complex, '(1,1)');
 complex_overlaps 
------------------
 t
(1 row)

SELECT complex_overlaps('(2,2)'::complex, '(3,2)');
 complex_overlaps 
------------------
 f
(1 row)

SELECT complex_mult( '(2,7)'::complex, '(3,9)');
 complex_mult 
--------------
 (-57,39)
(1 row)

SELECT complex_mult( '(3,2)'::complex, '(1,7)');
 complex_mult 
--------------
 (-11,23)
(1 row)

SELECT complex_mult( '(1,1)'::complex, '(1,1)');
 complex_mult 
--------------
 (0,2)
(1 row)

SELECT complex_polar( '(4,3)');
 complex_polar 
---------------
 (5,36.8699)
(1 row)

SELECT complex_xy('(5,36.8699)');
 complex_xy 
------------
 (4,3)
(1 row)

SELECT complex_xy(complex_polar( '(4,3)'));
 complex_xy 
------------
 (4,3)
(1 row)

SELECT complex_polar(complex_theta_add('(4,3)',20));
   complex_polar   
-------------------
 (7.31874,56.8699)
(1 row)

