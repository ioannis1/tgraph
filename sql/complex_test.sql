CREATE EXTENSION complex;

SET complex.style = 'none';

SELECT '(1,2)'::complex  ;
SELECT '1,2j''::complex  ;
SELECT '1''::complex     ;
SELECT '2j''::complex    ;
SELECT '1-2j''::complex  ;

SELECT '(1,2)'::complex +   '(1,2)';
SELECT '(1,2)'::complex -   '(1,2)';
SELECT '(1,2)'::complex =   '(1,2)';
SELECT '(1,2)'::complex >   '(0,2)';
SELECT '(1,2)'::complex >=  '(8,2)';

SELECT mag( '(3,4)'::complex );
SELECT complex_mag_squared( '(3,4)'::complex );

SELECT  2 > '(3,4)'::complex;
SELECT 25 >= '(3,4)'::complex;

SELECT  ('(1,8)'::complex@);      -- complex conjugate
SELECT |'(3,4)'::complex ;        -- magnitude

SELECT complex_new(1,8);
SELECT complex_new(3,1);

SELECT complex_real('(-2,3)'::complex);
SELECT complex_img('(2,-3)'::complex);

SELECT complex_overlaps('(2,2)'::complex, '(1,1)');
SELECT complex_overlaps('(2,2)'::complex, '(3,2)');

SELECT complex_mult( '(2,7)'::complex, '(3,9)');
SELECT complex_mult( '(3,2)'::complex, '(1,7)');
SELECT complex_mult( '(1,1)'::complex, '(1,1)');

SELECT complex_polar( '(4,3)');
SELECT complex_xy('(5,36.8699)');
SELECT complex_xy(complex_polar( '(4,3)'));
SELECT complex_polar(complex_theta_add('(4,3)',20));


