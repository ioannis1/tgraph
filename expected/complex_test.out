CREATE EXTENSION complex;
SELECT '(1,2')::complex;
ERROR:  syntax error at or near ")"
LINE 1: SELECT '(1,2')::complex;
                     ^
SELECT '(1,2)'::complex +   '(1,2)';
 ?column? 
----------
 (2,4)
(1 row)

SELECT '(1,2)'::complex -   '(1,2)';
 ?column? 
----------
 (0,0)
(1 row)

SELECT '(1,2)'::complex =   '(1,2)';
 ?column? 
----------
 t
(1 row)

SELECT '(1,2)'::complex >   '(0,2)';
 ?column? 
----------
 t
(1 row)

SELECT '(1,2)'::complex >=  '(8,2)';
 ?column? 
----------
 f
(1 row)

SELECT mag( '(3,4)'::complex );
 mag 
-----
   5
(1 row)

SELECT complex_mag_squared( '(3,4)'::complex );
 complex_mag_squared 
---------------------
                  25
(1 row)

SELECT  2 > '(3,4)'::complex;
 ?column? 
----------
 f
(1 row)

SELECT 25 >= '(3,4)'::complex;
 ?column? 
----------
 t
(1 row)

SELECT  ('(1,8)'::complex@);      -- complex conjugate
 ?column? 
----------
 (1,-8)
(1 row)

SELECT |'(3,4)'::complex ;        -- magnitude
 ?column? 
----------
        5
(1 row)

SELECT complex_new(1,8);
 complex_new 
-------------
 (1,8)
(1 row)

SELECT complex_new(3,1);
 complex_new 
-------------
 (3,1)
(1 row)

SELECT complex_real('(-2,3)'::complex);
 complex_real 
--------------
           -2
(1 row)

SELECT complex_img('(2,-3)'::complex);
 complex_img 
-------------
          -3
(1 row)

SELECT complex_overlaps('(2,2)'::complex, '(1,1)');
 complex_overlaps 
------------------
 t
(1 row)

SELECT complex_overlaps('(2,2)'::complex, '(3,2)');
 complex_overlaps 
------------------
 f
(1 row)

